packages/dropdownable/components/Dropdownable.tsx

```diff
@@ -0,0 +1,235 @@
+import React from "react";
+import styled, { css } from "react-emotion";
+
+import Overlay from "../../shared/components/Overlay";
+import DropdownContents from "./DropdownContents";
+
+export enum Direction {
+  BottomLeft = "bottom-left",
+  BottomRight = "bottom-right",
+  TopLeft = "top-left",
+  TopRight = "top-right"
+}
+
+const DEFAULT_DIRECTION_PREFERENCES = [
+  Direction.BottomLeft,
+  Direction.BottomRight,
+  Direction.TopRight,
+  Direction.TopLeft
+];
+
+export interface DropdownableProps {
+  open: boolean;
+  dropdown: React.ReactNode;
+  preferredDirections?: Direction[];
+  matchWidth?: boolean;
+  onClose?: () => void;
+}
+
+interface PositionCoord {
+  top: number;
+  left: number;
+}
+
+interface State {
+  direction: Direction;
+  position: PositionCoord;
+  dropdownWidth?: number;
+}
+
+const METHODS_TO_BIND = [
+  "getDropdown",
+  "setPosition",
+  "calculateBestDirection",
+  "dropdownStyles",
+  "positionForDirection",
+  "childBounds",
+  "windowDimensions",
+  "dropdownDimensions"
+];
+
+class Dropdownable extends React.Component<DropdownableProps, State> {
+  public static defaultProps: Partial<DropdownableProps> = {
+    preferredDirections: DEFAULT_DIRECTION_PREFERENCES,
+    matchWidth: false
+  };
+
+  private child = React.createRef<HTMLDivElement>();
+  private dropdown = React.createRef<HTMLDivElement>();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      direction: Direction.BottomLeft,
+      position: {
+        top: 0,
+        left: 0
+      }
+    };
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentDidMount() {
+    window.addEventListener("resize", () => this.setPosition());
```

Comment:
This might be a good place to use `requestAnimationFrame` to throttle how often we need to set the position.

I'm not seeing much of a difference in frame rate when the dropdown is open vs when it's closed, so I don't think this is a blocker.